sql/rename.go

```diff
@@ -83,7 +87,7 @@ func (p *planner) RenameTable(n *parser.RenameTable) (planNode, error) {
 		return nil, err
 	}
 
-	if n.Name.Table() == string(n.NewName) {
+	if n.Name.Table() == n.NewName.Table() {
```

Comment:
the `noop` case requires that the src/dst databases are also identical, so this is not correct